public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+
+  When the user clicks on a hero,
+  the component sets its `selectedHero` property which
+  is **bound to the `hero` input property** of the `HeroDetailComponent`.
+  The `HeroDetailComponent` detects the changed hero and re-sets its form
+  with that hero's data values.
+
+  A "Refresh" button clears the hero list and the current selected hero before refetching the heroes.
+  The `HeroListComponent` also has an `onSave` method that you'll wire up later in this guide.
+
+  The remaining `HeroListComponent` and `HeroService` implementation details are not relevant to understanding reactive forms.
+  The techniques involved are covered elsewhere in the documentation, including the _Tour of Heroes_
+  [here](../tutorial/toh-pt3.html "ToH: Multiple Components") and [here](../tutorial/toh-pt4.html "ToH: Services").
+
+  If you're coding along with the steps in this reactive forms tutorial, 
+  create the pertinent files based on the 
+  [source code displayed below](#source-code "Reactive Forms source code").
+  Then return here to learn about _form array_ properties.
+
+.l-main-section
+a#form-array
+:marked
+  ## Use _FormArray_ to present an array of _FormGroups_
+  So far, you've seen `FormControls` and `FormGroups`.
+  A `FormGroup` is a named object whose property values are `FormControls` and other `FormGroups`.
+
+  Sometimes you need to present an arbitrary number of controls or groups.
+  For example, a hero may have zero, one, or any number of addresses.
+  
+  The `Hero.addresses` property is an array of `Address` instances.
+  An _address_ `FormGroup` can display one `Address`.
+  An Angular `FormArray` can display an array of _address_ `FormGroups`.
+
+  To work with a `FormArray` you do the following:
+  1. Define the items (`FormControls` or `FormGroups`) in the array.
+  1. Initialize the array with items created from data in the _data model_.
+  1. Add and remove items as the user requires.
+
+  In this guide, you define a `FormArray` for `Hero.addresses` and
+  let the user add or modify addresses (removing addresses is your homework).
+
+  You’ll need to redefine the form model in the `HeroDetailComponent` constructor,
+  which currently only displays the first hero address in an _address_ `FormGroup`. 
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'address-form-group')(format=".")
+
+:marked
+  ### From _address_ to _secret lairs_
+
+  From the user's point of view, heroes don't have _addresses_.
+  _Addresses_ are for mere mortals. Heroes have _secret lairs_!
+  Replace the _address_ `FormGroup` definition with a _secretLairs_ `FormArray` definition:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'secretLairs-form-array')(format=".")
+
+.alert.is-helpful
+  :marked
+    Changing the form control name from `address` to `secretLairs` drives home an important point:
+    the _form model_ doesn't have to match the _data model_.
```

Comment:
My course has not yet been published. It should go out this week. (It got stuck behind the backlog over the holiday.) But I do think that when people here "model" they are thinking "data model". So it is important to be clear on the difference between the data managed by the NgForm ("form model") and the data managed by the app ("data model").